src/Development/IDE/Types/Options.hs

```diff
@@ -25,12 +23,6 @@ data IdeOptions = IdeOptions
   { optPreprocessor :: GHC.ParsedSource -> IdePreprocessedSource
     -- ^ Preprocessor to run over all parsed source trees, generating a list of warnings
     --   and a list of errors, along with a new parse tree.
-  , optGhcSession :: IO (FilePath -> Action HscEnvEq)
```

Comment:
Instead of removing this can we simply change it to `Action (FilePath -> Action HscEnvEq)`? By removing this you are turning an explicit dependency into an implicit one (if I don’t define the rule, things will explode).

Reply:
Added back as you suggested. That allowed me to reduce the number of changes in this PR. And it also removed the `dir` argument of the `loadGhcSessionIO` rule. Thanks!